src/widget/OktaSignIn.js

```diff
@@ -94,6 +111,8 @@ var OktaSignIn = (function() {
       hide,
       show,
       remove,
+      before,
```

Comment:
Why would you think of 2 ways to add hooks rather only via config?
Are you anticipating hooks may be added on the fly?

Reply:
I was thinking more of the case where config may be reconstituted from a string like an environment variable or yaml file, and the developer does not want to include any functions. Current logic (success function) is added at runtime, not config.

Reply:
I'd leaning toward single way to do the job rather being flexible until developer explicitly asks. just by `0.02`.

Reply:
I would prefer to have the flexibility inbuilt rather than later when someone ask about it.. also it doesn't seem much and nicely meld with existing `hide` `show` functions